sink_windows_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//go:build windows
+
+package zap
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWindowsPaths(t *testing.T) {
+	sr := newSinkRegistry()
+
+	var openFilename string
+	sr.openFile = func(filename string, _ int, _ os.FileMode) (*os.File, error) {
+		openFilename = filename
+		return nil, assert.AnError
```

Comment:
Not a blocker, but a function shared between subtests that modifies its lexical scope smells a bit.

How do you feel about pushing this setup inside the test logic as well:

```
t.Run(..., func(t *testing.T) {
  sr := newSinkRegistry()

  var openFilename string
  sr.openFile = func(...) (*os.File, error) {
    openFilename = filename
    return nil, assert.AnError
  }

  // ...

  assert.Equal(t, tt.path, openFilename, ...)
})
```